bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/AnnotationActionHandler.java

```diff
@@ -69,7 +70,13 @@ public AnnotationActionHandler(Action module, ActionType mt, Method method, Obje
                 if (annotationsOnParam[0] instanceof ActionInput inputAnnotation) {
                     // check if the moduleType has a configdescription with this input
                     if (hasInput(moduleType, inputAnnotation.name())) {
-                        args.add(i, context.get(inputAnnotation.name()));
+                        Object value = context.get(inputAnnotation.name());
+                        // fallback to configuration as this is where the UI stores the input values
+                        if (value == null) {
+                            value = SerialisedInputsToActionInputs.map(moduleType, moduleType.getInputs().get(i),
+                                    module.getConfiguration().get(inputAnnotation.name()), null);
+                        }
```

Comment:
@florian-h05 : you added these lines but to be honnest with you I don't understand in which conditions this is used. As unit provider is not provided in this case, I would like to understand the potential impact.

PS: I am going to add a log to see if this is used when I use the REST API but in principle it should not be required for this usage.

Reply:
UnitProvider should be used in this case as well.
This change is required to make Thing actions usable in UI-based rules.

Reply:
If I used null, that means UnitProvider was not available...

Reply:
Can you change that? Or have you already changed it? I cannot have a look at the code as I’m currently away from keyboard …

Reply:
Not yet.
I have to think how I could retrieve this service in that class...

Reply:
This is a problem only for QuantityType inputs.

By the way, I hav an improvement to apply, unitProvider is not required in case the input is a string containing the number AND a unit.

Reply:
The current workaround in Main UI is that you create a string containing the value and the unit when calling the action, instead of just providing a number.

Reply:
By using a Component as suggested by @J-N-K , the problem with unit provider is solved.

Reply:
UnitProvider will now always be available with the last changes and the introduction of an OSGi component.